### PR TITLE
Update weather on applicationDidBecomeActive(_:)

### PR DESCRIPTION
**Pull request by sharplet**
I've been using Tropos with the "when in use" location authorization, and finding that sometimes I would launch the app and no weather update would occur. I _think_ the cause might be that `applicationWillEnterForeground(_:)` is called while the app is in a background state, causing the location check to fail / never complete.

Users with "always" location authorization and background fetch enabled will still get more timely weather updates.

I added some error logging, and also rewrote the app delegate in Swift, because why not!

### CODE REVIEW

File: Tropos/Sources/Controllers/TRSettingsController.m [hunk 1]
```diff
@@ -1,75 +0,0 @@
-#import "TRSettingsController.h"
-#import "NSBundle+TRBundleInfo.h"
-
-NSString *const TRSettingsDidChangeNotification = @"TRSettingsDidChangeNotification";
-static NSString *const TRSettingsUnitSystemKey = @"TRUnitSystem";
-static NSString *const TRSettingsLastVersionKey = @"TRLastVersion";
-
-@interface TRSettingsController ()
```
Comment (sharplet) [inline Tropos/Sources/Controllers/TRSettingsController.m]: This was a duplicate of `SettingsController` as defined in the Core framework.

File: Tropos/Sources/main.m [hunk 2]
```diff
@@ -1,8 +0,0 @@
-#import "TRAppDelegate.h"
```
Comment (jamescmartinez) [inline Tropos/Sources/main.m]: Did you mean to remove this?

File: Tropos/Sources/Controllers/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+@UIApplicationMain
+final class AppDelegate: UIResponder, UIApplicationDelegate {
+    let applicationController = ApplicationController()
+    var window: UIWindow?
+
+    func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions _: [UIApplicationLaunchOptionsKey: Any]? = nil
+    ) -> Bool {
```
Comment (jamescmartinez) [inline Tropos/Sources/Controllers/AppDelegate.swift]: This function is _giant_. Could it be broken down?

File: Tropos/Sources/main.m (hunk 2 above)
Comment (sharplet) [reply Tropos/Sources/main.m]: Yep, this is superseded by the `@UIApplicationMain` annotation in `AppDelegate.swift`.

File: Tropos/Sources/Controllers/AppDelegate.swift (hunk 3 above)
Comment (sharplet) [reply Tropos/Sources/Controllers/AppDelegate.swift]: Sure, we could extract some setup functions.

Comment (sharplet) [reply Tropos/Sources/Controllers/AppDelegate.swift]: Extracted `setupHockey()` in 489f7f1

File: Core/Specs/Models/WeatherUpdateCacheSpec.swift [hunk 4]
```diff
@@ -5,26 +5,31 @@ import Nimble
 private let testCacheFileName = "TestWeatherUpdate"
 
 private var testCachesDirectory: URL {
-    return FileManager.default
-        .urls(for: .documentDirectory, in: .userDomainMask)
-        .first!
+    return try! FileManager.default
```
[the remark this answers is not in the document]
Comment (sharplet) [reply Core/Specs/Models/WeatherUpdateCacheSpec.swift]: Disabled the warning for this case.

**Comment by sharplet** [comment]
ugh, finally!

**Pull request merged**